pydeseq2/DeseqStats.py

```diff
@@ -191,6 +191,15 @@ def __init__(
         self.batch_size = batch_size
         self.joblib_verbosity = joblib_verbosity
 
+        # If the `refit_cooks` attribute of the dds object is True, check that outliers
+        # were actually refitted.
+        if dds.refit_cooks:
+            assert hasattr(dds, "replaced"), (
+                "dds has 'refit_cooks' set to True but Cooks outliers have not been "
+                "refitted. Please run 'dds.refit()' first or set 'dds.refit_cooks' "
+                "to False."
+            )
```

Comment:
Instead of an `AssertionError`, what about raising an `AttributeError`? Otherwise LGTM

Reply:
Good idea, an `AttributeError` might be more appropriate indeed